vdr/vdr.go

```diff
@@ -165,58 +167,5 @@ func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, _ *t
 }
 
 func isDeactivated(document *did.Document) bool {
-	return len(document.Controller) == 0 && len(document.Authentication) == 0
-}
-
-// resolveControllers accepts a list of documents and finds their controllers
-// The resulting list are documents who control themselves
-func (r *VDR) resolveControllers(input []did.Document) ([]did.Document, error) {
-	// end of the chain
-	if len(input) == 0 {
-		return input, nil
-	}
-
-	var leaves []did.Document
-	var refsToResolve []did.DID
-	var nodes []did.Document
-
-	// for each input document, find its controllers or add the doc itself if its controls itself
-	for _, doc := range input {
-		if len(doc.Controller) == 0 && len(doc.Authentication) > 0 {
-			// no controller -> doc is its own controller
-			leaves = append(leaves, doc)
-			continue
-		}
-		for _, ctrlDID := range doc.Controller {
-			if doc.ID.Equals(ctrlDID) {
-				if len(doc.Authentication) > 0 {
-					// doc is its own controller
-					leaves = append(leaves, doc)
-				}
-			} else {
-				// add did to be resolved later
-				refsToResolve = append(refsToResolve, ctrlDID)
-			}
-		}
-	}
-	// resolve all unresolved docs
-	// TODO: check for recursions in controllers. Behaviour must be described in spec:
-	// nuts-foundation/nuts-specification#39
-	for _, ref := range refsToResolve {
-		node, _, err := r.store.Resolve(ref, nil)
-		if err != nil {
-			return nil, fmt.Errorf("unable to resolve controllers: %w", err)
-		}
-		nodes = append(nodes, *node)
-	}
-	newLeaves, err := r.resolveControllers(nodes)
-	if err != nil {
-		return nil, err
-	}
-	// Merge local leaves and new found leaves
-	for _, leave := range newLeaves {
-		leaves = append(leaves, leave)
-	}
-
-	return leaves, nil
+	return len(document.Controller) == 0 && len(document.CapabilityInvocation) == 0
```

Comment:
duplicate func? (also in memoryStore)

Reply:
Yes, dependencies are a bit of a problem here. We could remove the check from the store and only do it in the VDR?

Reply:
moved it to a func in `store`